site/content/how-to/traffic-management/response-headers.md

```diff
@@ -0,0 +1,199 @@
+
+
+The [gateway](https://gateway-api.sigs.k8s.io/api-types/gateway/) resource is typically deployed by the [cluster operator](https://gateway-api.sigs.k8s.io/concepts/roles-and-personas/#roles-and-personas_1). To deploy the gateway:
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: gateway
+spec:
+  gatewayClassName: nginx
+  listeners:
+  - name: http
+    port: 80
+    protocol: HTTP
+EOF
+```
+
+### Configure the basic HTTPRoute
+
+This gateway defines a single listener on port 80. Since no hostname is specified, this listener matches on all hostnames.
+
+The HTTPRoute is typically deployed by the [application developer](https://gateway-api.sigs.k8s.io/concepts/roles-and-personas/#roles-and-personas_1). To deploy the `headers` HTTPRoute:
+
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: headers
+spec:
+  parentRefs:
+  - name: gateway
+    sectionName: http
+  hostnames:
+  - "cafe.example.com"
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /headers
+    backendRefs:
+    - name: headers
+      port: 80
+EOF
+```
+
+### Send Traffic to the Basic Headers Application
+
+To access the application, we will use `curl` to send requests to the `headers` endpoint. Notice our configured header values can be seen in the `responseHeaders` section below. We have four custom response headers defined - `X-Header-Unmodified`, `X-Header-Add`, `X-Header-Set`, `X-Header-Remove`.
+
+```shell
+curl -i --resolve cafe.example.com:$GW_PORT:$GW_IP http://cafe.example.com:$GW_PORT/headers
+```
+
+```text
+HTTP/1.1 200 OK
+Server: nginx/1.25.5
+Date: Mon, 06 May 2024 19:08:39 GMT
+Content-Type: text/plain
+Content-Length: 2
+Connection: keep-alive
+X-Header-Unmodified: unmodified
+X-Header-Add: add-to
+X-Header-Set: overwrite
+X-Header-Remove: remove
+
+ok
+```
+
+### Configure the HTTPRoute with Response Header Modifiers
+
+Let's configure the HTTPRoute with `ResponseHeaderModifier` filter:
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: headers
+spec:
+  parentRefs:
+  - name: gateway
+    sectionName: http
+  hostnames:
+  - "cafe.example.com"
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /headers
+    filters:
+    - type: ResponseHeaderModifier
+      responseHeaderModifier:
+        set:
+        - name: X-Header-Set
+          value: overwritten-value
+        add:
+        - name: X-Header-Add
+          value: this-is-the-appended-value
+        remove:
+        - X-Header-Remove
+    backendRefs:
+    - name: headers
+      port: 80
+EOF
+```
+
+This HTTPRoute has a few important properties:
+
+- The `parentRefs` references the gateway resource that we created, and specifically defines the `http` listener to attach to, via the `sectionName` field.
```

Comment:
In this section, we want to highlight the response header filter. The other properties of the HTTPRoute can be moved under the original HTTPRoute.  Then, here we will just emphasize that we've added the ResponseHeaderModifier filter and describe what it does.